IdentityCore/src/util/NSDictionary+MSIDExtensions.m

```diff
@@ -111,5 +111,56 @@ - (NSDictionary *)dictionaryByRemovingFields:(NSArray *)fieldsToRemove
     return array;
 }
 
+- (BOOL)assertType:(Class)type
+           ofField:(NSString *)field
+           context:(id <MSIDRequestContext>)context
+         errorCode:(NSInteger)errorCode
+             error:(NSError **)error;
+{
+    id fieldValue = self[field];
+    if (![fieldValue isKindOfClass:type])
+    {
+        __auto_type message = [NSString stringWithFormat:@"%@ is not a %@.", field, type];
+        
+        if (error)
+        {
+            *error = MSIDCreateError(MSIDErrorDomain,
+                                     errorCode,
+                                     message,
+                                     nil,
+                                     nil, nil, context.correlationId, nil);
+        }
+        
+        MSID_LOG_ERROR(nil, @"%@", message);
+        return NO;
+    }
+    
+    return YES;
+}
+
+- (BOOL)assertContainsField:(NSString *)field
+                    context:(id <MSIDRequestContext>)context
+                      error:(NSError **)error
+{
+    id fieldValue = self[field];
+    if (!fieldValue)
+    {
+        __auto_type message = [NSString stringWithFormat:@"%@ is missing.", field];
+        
+        if (error)
+        {
+            *error = MSIDCreateError(MSIDErrorDomain,
+                                     MSIDErrorServerInvalidResponse,
+                                     message,
+                                     nil,
+                                     nil, nil, context.correlationId, nil);
+        }
+        
+        MSID_LOG_ERROR_PII(nil, @"%@", message);
```

Comment:
I don't think field name needs to be PII

Reply:
done